### PR TITLE
Refactoring Event manager invoke target listener for event

### PR DESCRIPTION
**Pull request by marchant**

### CODE REVIEW

File: core/event/event-manager.js [hunk 1]
```diff
@@ -2259,6 +2228,19 @@ var EventManager = exports.EventManager = Montage.specialize(/** @lends EventMan
         }
     },
 
+	_invokeTargetListenerForEvent: {
+		value: function(iTarget, jListener,mutableEvent,identifierSpecificPhaseMethodName,phaseMethodName) {
```
Comment (rayshan) [inline core/event/event-manager.js]: space after function & b/t args

File: core/event/event-manager.js [hunk 2]
```diff
@@ -2259,6 +2228,19 @@ var EventManager = exports.EventManager = Montage.specialize(/** @lends EventMan
...
+			var listenerFunction;
+			if ((identifierSpecificPhaseMethodName && typeof (listenerFunction = jListener[identifierSpecificPhaseMethodName]) === FUNCTION_TYPE)
+			 || (typeof (listenerFunction = jListener[phaseMethodName]) === FUNCTION_TYPE)
+			 || (typeof (listenerFunction = jListener.handleEvent) === FUNCTION_TYPE)) {
+ 				listenerFunction.call(jListener, mutableEvent);
+            } else if (typeof jListener === FUNCTION_TYPE) {
```
Comment (rayshan) [inline core/event/event-manager.js]: too little indent

**Comment by rayshan** [comment]
LGTM, relevant tests passed, can you please squash your commits?

This file has a lot of whitespace & indent issues in general, I have a local copy that fixes them, can submit it later.

**Comment by rayshan** [comment]
Per discussion with @marchant, this is ready to be committed now that it's squashed. I'll have a separate PR that fixes all the whitespace issues in `event-manager.js`.

**Comment by thibaultzanini** [comment]
Looks good to me.

**Comment by rayshan** [comment]
thank you sir

**Pull request merged**